src/scry/analyzer.cr

```diff
@@ -5,12 +5,22 @@ require "./publish_diagnostic"
 
 module Scry
   struct Analyzer
+    @@source = [] of String
+
     def initialize(@workspace : Workspace, @text_document : TextDocument)
       @diagnostic = PublishDiagnostic.new(@workspace, @text_document.uri)
     end
 
     def run
-      @text_document.text.map { |t| analyze(t) }.flatten.uniq
+      if @@source != @text_document.text
+        @@source = @text_document.text
+        @@source.map do |text|
+          unless @text_document.filename.starts_with?("/usr/lib") ||
```

Comment:
Is this a global shard home directory? Is there an env var we could use instead?